src/Raven.Server/Documents/Queries/QueryBuilder.cs

```diff
@@ -36,6 +36,12 @@ public static class QueryBuilder
     {
         private static readonly KeywordAnalyzer KeywordAnalyzer = new KeywordAnalyzer();
 
+        private static readonly MultiTermQuery.ConstantScoreAutoRewrite _secondaryBetweenRewriteMethod = new MultiTermQuery.ConstantScoreAutoRewrite
+        {
+            TermCountCutoff = BooleanQuery.MaxClauseCount, // can be set using Query.MaxClauseCount
```

Comment:
we can change that in the config, shouldn't we take that from there?

Reply:
We are doing that. The reason this is done this way is that this is the limit _inside lucene_

Reply:
My bad, now I got this. For future reference: This assumes that ServerStore (the place where we are setting this) is initialized faster than QueryBuilder.

Reply:
That must be the case, we cannot create a query builder without creating a db, which require a server